nslsii/iocs/lakeshore_control.py

```diff
@@ -0,0 +1,240 @@
+                                 name='}}Gain:I-SP')
+
+    pid_derivative_rb = pvproperty(value=_pid_derivative_val,
+                                   read_only=True,
+                                   dtype=ChannelType.DOUBLE,
+                                   name='}}Gain:D-RB')
+    pid_derivative_sp = pvproperty(value=_pid_derivative_val,
+                                   dtype=ChannelType.DOUBLE,
+                                   name='}}Gain:D-SP')
+
+    # output parameters
+
+    _out_current_val = 0.
+    _out_man_current_val = 0.
+    _out_max_current_val = 0.
+    _out_resistance_val = 0.
+
+    out_current = pvproperty(value=_out_current_val,
+                             dtype=ChannelType.DOUBLE,
+                             name='}}Out-I')
+
+    out_man_current_rb = pvproperty(value=_out_man_current_val,
+                                    read_only=True,
+                                    dtype=ChannelType.DOUBLE,
+                                    name='}}Out:Man-RB')
+    out_man_current_sp = pvproperty(value=_out_man_current_val,
+                                    dtype=ChannelType.DOUBLE,
+                                    name='}}Out:Man-SP')
+
+    out_max_current_rb = pvproperty(value=_out_max_current_val,
+                                    read_only=True,
+                                    dtype=ChannelType.DOUBLE,
+                                    name='}}Out:MaxI-RB')
+    out_max_current_sp = pvproperty(value=_out_max_current_val,
+                                    dtype=ChannelType.DOUBLE,
+                                    name='}}Out:MaxI-SP')
+
+    out_resistance_rb = pvproperty(value=_out_resistance_val,
+                                   read_only=True,
+                                   dtype=ChannelType.DOUBLE,
+                                   name='}}Out:R-RB')
+    out_resistance_sp = pvproperty(value=_out_resistance_val,
+                                   dtype=ChannelType.DOUBLE,
+                                   name='}}Out:R-SP')
+
+    # Putter/Getter Methods
+
+    @setpoint.putter
+    async def setpoint(self, instance, value):
+
+        # select channel
+        prefix = self.ioc.prefix.replace('{', '{'*2)
+        channel = self._target_channel_val
+        t_k = f'{prefix}-Chan:{channel}'
+        if t_k in self.ioc.groups:
+            pass
+        else:
+            return instance.value
+        t_v = self.ioc.groups[t_k]
+
+        # apply cmd
+        indx = self._indx
+        cmd = f'{value},{indx}'
+        await t_v.cmd.write(value=cmd)
+
+        self._rb_val = value
+        return value
+
+    @done.getter
+    async def done(self, instance):
+        return self._done_val
+
+    @target_channel.getter
+    async def target_channel(self, instance):
+        return self._target_channel_val
+
+    @target_channel.putter
+    async def target_channel(self, instance, value):
+        self._target_channel_val = value
```

Comment:
Additional attributes for these should be unnecessary; you could replace:
`self._target_channel_val` with `self.target_channel.value` (for reading) or `await self.target_channel.write(value=0)` for writing.

Reply:
The extra layer was added to implement potential extensions for simulating hardware issues.